src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -126,11 +131,52 @@ std::shared_ptr<uint32_t> GetCounterHolder(const char * name)
     return std::make_shared<uint32_t>(0);
 }
 
+bool IsSRPType(const char * domain)
+{
+    return strcmp(kSRPDot, domain) == 0;
+}
+
 } // namespace
 
 namespace chip {
 namespace Dnssd {
 
+/**
+ * @brief Callback that is called when the timeout for resolving on the kSRPDot domain has expired.
+ *
+ * @param[in] systemLayer The system layer.
+ * @param[in] callbackContext The context passed to the timer callback.
+ */
+void SRPTimerExpiredCallback(System::Layer * systemLayer, void * callbackContext)
```

Comment:
Why are these global functions?  Shouldn't they be in the anonymous namespace above?  Or inside the one below, either way.  But they need to be in an anonymous namespace.

Reply:
The CancelTimer is called by the DnssdContexts and i can't call it from there if I put it in an anonymous namespace

Reply:
OK, but you can put SRPTimerExpiredCallback, StartSRPTimer in an anonymous namespace.

And CancelSRPTimer should just be a private member function on ResolveContext, not a global function.

Reply:
done.